client/control/components/Header.js

```diff
@@ -32,4 +52,56 @@ Header.propTypes = {
   currentLocation: pt.string.isRequired,
   routes: pt.array.isRequired,
   params: pt.object.isRequired,
+  token: pt.shape({ key: pt.string.isRequired }),
+  user: pt.object.isRequired,
 };
+
+export default connect(
+  (state, ownProps) => ({
+    ...ownProps,
+    token: state.auth.token,
+  }),
+)(Header);
+
+
+class APITokenExpander extends Component {
+  static propTypes = {
+    token: pt.shape({ key: pt.string }),
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = { expanded: false };
+    this.toggleExpand = ::this.toggleExpand;
+  }
+
+  toggleExpand() {
+    this.setState(state => ({ ...state, expanded: !state.expanded }));
```

Comment:
You don't need the `...state` operator, as React will create the new state object on its own. E.g.

```this.setState(state => ({ ...state, expanded: !state.expanded }));```
is equivalent to
```this.setState({ expanded: !state.expanded });```
(especially in this case, where `expanded` is the only property in state)